mu/logic.py

```diff
@@ -855,6 +856,88 @@ def connect_to_status_bar(self, status_bar):
             for mode in self.modes.values():
                 device_selector.device_changed.connect(mode.device_changed)
 
+    def detect_theme_preference(self):
+        """Detect OS dark/light theme preference.
+        Currently implemented for Mac, Windows, GNOME. For any other
+        operating systems, it will always return "day"
+        """
+        # Implementations inspired by the following Stackoverflow submissions:
+        # https://stackoverflow.com/questions/65294987/detect-os-dark-mode-in-python
+
+        def macos_prefers_dark_theme(self):
+            if sys.platform != "darwin":
+                cmd = "defaults read -g AppleInterfaceStyle"
+                p = subprocess.Popen(
+                    cmd,
+                    stdout=subprocess.PIPE,
+                    stderr=subprocess.PIPE,
+                    shell=True,
+                )
+                return bool(p.communicate()[0])
+
+        def windows_prefers_dark_theme(self):
+            if sys.platform != "win32":
+                return False
+            try:
+                import winreg
+            except ImportError:
+                return False
+            registry = winreg.ConnectRegistry(None, winreg.HKEY_CURRENT_USER)
+            reg_keypath = (
+                r"SOFTWARE\Microsoft\Windows\CurrentVersion\Themes\Personalize"
+            )
+            try:
+                reg_key = winreg.OpenKey(registry, reg_keypath)
+            except FileNotFoundError:
+                return False
+
+            for i in range(1024):
+                try:
+                    value_name, value, _ = winreg.EnumValue(reg_key, i)
+                    if value_name == "AppsUseLightTheme" and value == 0:
+                        return True
+                except OSError:
+                    break
+            return False
+
+        def GNOME_prefers_dark_theme(self):
+            if sys.environ["XDG_CURRENT_DESKTOP"] != "GNOME":
+                return False
+            # Detect dark mode in GNOME
+            getArgs = [
+                "gsettings",
+                "get",
+                "org.gnome.desktop.interface",
+                "gtk-theme",
+            ]
+
+            currentTheme = (
+                subprocess.run(getArgs, capture_output=True)
+                .stdout.decode("utf-8")
+                .strip()
+                .strip("'")
+            )
+
+            darkIndicator = "-dark"
+            return currentTheme.endswith(darkIndicator)
```

Comment:
Not sure if `darkIndicator` is needed. Why not just `return currentTheme.endswith("-dark")` ..?